doc/development/deprecations.rst

```diff
@@ -34,6 +34,20 @@ Pending deprecations
 
   - Added and deprecated for ``Sum`` and ``Prod`` instances in v0.35
 
+* Accessing ``qml.ops.Hamiltonian`` with new operator arithmetic is deprecated. Using ``qml.Hamiltonian``
+  with new operator arithmetic enabled now returns a ``LinearCombination`` instance. Some functionality
```

Comment:
Do we want to hide the fact that the new returned operator is `LinearCombination` from users?

```suggestion
* Accessing ``qml.ops.Hamiltonian`` with new operator arithmetic is deprecated. Some functionality
```

Reply:
I agree, the user probably doesn't have to know what a `LinearCombination`, or that it is part of new opmath.

Reply:
Let's keep mentioning `LinearCombination` here. One thing we might want to move away from in the docs and QML demos is using "Hamiltonian" out of context. That is, we will talk about Hamiltonians as observables determining energy in a system, but in other cases it might suffice to talk about "linear combinations".